### PR TITLE
Grant multiple permissions

### PR DESCRIPTION
**Pull request by icemac**
Add support to grant multiple permissions with one ZCML statement.

### CODE REVIEW

**Comment by mgedmin** [comment]
I don't want to review a PR that mixes PEP-8 reformatting with actual changes.

**Comment by jinty** [comment]
On Thu, Oct 13, 2016 at 11:05:36AM -0700, Marius Gedminas wrote:
> I don't want to review a PR that mixes PEP-8 reformatting with actual changes.

+1

Being a firm believer in "Do not PEP 8 unto others; only PEP 8 thy self."

-- 
Brian Sutherland

**Comment by thet** [comment]
I disagree. In community driven projects, following coding standards like pep 8 is all about readability.

There are two commits. One for pep8, and this one is about the "Grant multiple permissions" feature:
https://github.com/zopefoundation/zope.securitypolicy/pull/5/commits/14a993be03f447c53b8726bcd9c01fae15825e5d

File: src/zope/securitypolicy/metaconfigure.py [hunk 1]
```diff
@@ -26,37 +26,43 @@
...
-                  (permission is not None))
+                  (permission is not None) +
+                  (permissions is not None))
 
     if nspecified != 2:
         raise ConfigurationError(
-            "Exactly two of the principal, role, and permission attributes "
-            "must be specified")
+            "Exactly two of the principal, role, and permission resp. "
+            "permissions attributes must be specified")
```
Comment (thet) [inline src/zope/securitypolicy/metaconfigure.py]: Does not sound clear enough to me. I'm not a native english speaker, but what about this: "Exactlty two out of these arguments must be specified: principal, role and permission or permissions."?

Comment (thet) [inline src/zope/securitypolicy/metaconfigure.py]: Ok, the wording was the same before. IMO no change needed.

**Comment by thet** [comment]
Regarding the pep 8 commit - is there anywhere a zope code style guideline?
At Plone, we have worked out a guideline, which works very well for us and has good tool support: http://docs.plone.org/develop/styleguide/python.html

What's missing in my opinion in the pep 8 commit of this PR is:
- One import per line,
- Alphabetically sorted imports (case insensitive),
- UTF 8 headers

And I would remove the distracting license header from the source files and move that into a ``LICENSE.rst`` file in the top level. I don't know if the zopefoundation policy still requires them in all source files?

However, these remarks are out of scope of this PR can be ignored for merging the pull request.

Anyone else giving a second review and merge then?

**Comment by mgedmin** [comment]
PEP-8 is *fine*, I just want it to be separate.  And PR #4 already exists, and is (mostly) good, and can be merged right now.  And once that's done, this PR's diffs will become readable again.

I've a policy of not merging other people's PRs if they have the commit bit themselves (especially if I give suggestions for possible changes, like I did on #4).

**Comment by icemac** [comment]
Replaced by #6.

**Comment by mgedmin** [comment]
FWIW github's pull requests can deal with rebases and forced pushes just fine, there's no need to close and open a new PR.  (Unless you find closing and opening a new PR easier for you, of course!)

**Pull request closed without merging**